autorest/codegen/serializers/general_serializer.py

```diff
@@ -20,12 +20,27 @@ def serialize_pkgutil_init_file(self) -> str:
 
     def serialize_init_file(self) -> str:
         template = self.env.get_template("init.py.jinja2")
-        return template.render(code_model=self.code_model, async_mode=self.async_mode,)
+        return template.render(code_model=self.code_model, async_mode=self.async_mode)
+
+    def _correct_credential_parameter(self):
+        credential_param = [
+            gp for gp in self.code_model.global_parameters.parameters if isinstance(gp.schema, CredentialSchema)
+        ][0]
+        credential_param.schema = CredentialSchema(async_mode=self.async_mode)
+        # we know that if there's a credential global param, it's always the first because
+        # we insert it there
+        self.code_model.global_parameters[0] = credential_param
```

Comment:
You don't need this line, since you change `credential_param.schema` by reference